src/main/java/bio/terra/service/dataset/DatasetService.java

```diff
@@ -486,6 +488,31 @@ public List<BulkLoadHistoryModel> getLoadHistory(
     }
   }
 
+  /**
+   * @param dataset the dataset to configure the AzureDataSourceFor
+   * @param userRequest the user making the request
+   * @return the name of the datasource created
+   * @throws RuntimeException when the external datasource could not be configured
+   */
+  public String getOrCreateExternalAzureDataSource(
```

Comment:
There's an existing method with the same name in `AzureSynapsePdao` that is used for both this and snapshot azure configuration. I think that because this is specific to the Dataset implementation, it probably belongs here? However, I could refactor the method in `AzureSynapsePdao` to take implementation independent pieces, and generate the credName, datasourceName, and metadataUrl itself.

Reply:
I started to do that and then it gets complicated, because some of the derived information is important for later parts of the method.

Reply:
Nevermind, I can move a lot of it, but I can't pull the accessInfo piece because the logic is different between snapshots and datasets